tomcat-8.5/src/org/dogtagpki/tomcat/JSSImplementation.java

```diff
@@ -19,21 +19,30 @@
 
 package org.dogtagpki.tomcat;
 
+import javax.net.ssl.SSLSession;
+
 import org.apache.tomcat.util.net.SSLHostConfig;
 import org.apache.tomcat.util.net.SSLHostConfigCertificate;
 import org.apache.tomcat.util.net.SSLUtil;
-import org.apache.tomcat.util.net.jsse.JSSEImplementation;
+import org.apache.tomcat.util.net.SSLSupport;
+import org.apache.tomcat.util.net.SSLImplementation;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class JSSImplementation extends JSSEImplementation {
+public class JSSImplementation extends SSLImplementation {
 
     public static Logger logger = LoggerFactory.getLogger(JSSUtil.class);
 
     public JSSImplementation() {
         logger.debug("JSSImplementation: instance created");
     }
 
+    @Override
+    public SSLSupport getSSLSupport(SSLSession session) {
+        logger.debug("JSSImplementation.getSSLSupport()");
+        return null;
```

Comment:
It isn't immediately clear what this is supposed to return or how it is useful. But Tomcat's OpenSSL adapter implementation is [here](https://github.com/apache/tomcat/blob/master/java/org/apache/tomcat/util/net/openssl/OpenSSLImplementation.java#L31). It looks like it might be a generic method instead.